src/readers/src/T2D.h

```diff
@@ -0,0 +1,240 @@
+#pragma once
+#include <istream>
+#include <numeric>
+#include <vector>
+
+namespace {
+template <typename T>
+class SerializableVec {
+ public:
+	SerializableVec() = default;
+
+	SerializableVec& set_size(size_t size) {
+		elem_count_ = size;
+		data_.reserve(elem_count_);
+		return *this;
+	}
+
+	friend std::istream& operator>>(std::istream& in, SerializableVec& vec) {
+		for (size_t i = 0; i < vec.elem_count_; ++i) {
+			T value;
+			in >> value;
+			vec.data_.push_back(value);
+		}
+		return in;
+	}
+	[[nodiscard]] const std::vector<T>& data() const { return data_; }
+
+ private:
+	size_t elem_count_{0};
+	std::vector<T> data_;
+};
+
+struct BodySize {
+	float x, y, z;  //размеры корпуса по осям x, y, z:
+
+	friend std::istream& operator>>(std::istream& in, BodySize& size) {
+		in >> size.x >> size.y >> size.z;
+		return in;
+	}
+};
+
+struct BodyParams {
+	float ZP;  // – высота бокса,
+	float TP;  // – толщина стенок бокса,
+	float TL;  // – толщина крышки бокса,
+	float BP;  // – z-координата дна бокса
+
+	friend std::istream& operator>>(std::istream& in, BodyParams& params) {
+		in >> params.ZP >> params.TP >> params.TL >> params.BP;
+		return in;
+	}
+};
+
+struct CoordsMinMax {
+	float min_x;
+	float min_y;
+	float max_x;
+	float max_y;
+
+	friend std::istream& operator>>(std::istream& in, CoordsMinMax& coords) {
+		in >> coords.min_x >> coords.min_y >> coords.max_x >> coords.max_y;
+		return in;
+	}
+};
+
+struct Coords {
+	float x;
+	float y;
+
+	friend std::istream& operator>>(std::istream& in, Coords& coords) {
+		in >> coords.x >> coords.y;
+		return in;
+	}
+};
+
+struct Spheres {
+	explicit Spheres(size_t count) {
+		spheres_centers_ = SerializableVec<Coords>{}.set_size(count);
+	}
+	float diameter_{0};
+	SerializableVec<Coords> spheres_centers_;
```

Comment:
1) shall we make it private with getter?
2) if no pleas return "_" at the end, we use it for private members only